src/Storage/Local.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Local storage adapter.
+ *
+ * @package SatisPress
+ * @license GPL-2.0-or-later
+ * @since 0.3.0
+ */
+
+declare ( strict_types = 1 );
+
+namespace SatisPress\Storage;
+
+use function SatisPress\send_file;
+use DirectoryIterator;
+use WP_Error;
+
+/**
+ * Local storage adapter class.
+ *
+ * @since 0.3.0
+ */
+class Local implements StorageInterface {
+	/**
+	 * Base directory.
+	 *
+	 * @var string
+	 */
+	protected $base_directory = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param string $base_directory Base storage directory.
+	 */
+	public function __construct( $base_directory ) {
+		$this->set_base_directory( $base_directory );
+	}
+
+	/**
+	 * Retrieve the hash value of the contents of a file.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param string $algorithm Algorithm.
+	 * @param string $file      Relative file path.
+	 * @return string|null
+	 */
+	public function checksum( $algorithm, $file ): string {
```

Comment:
How is determining the checksum specifically related to where a file is stored (here, Local)?

Reply:
Because where the file is stored determines what methods are available for deriving a checksum.

For instance, AWS S3 doesn't have an API for hashing files, so it needs to be stored in the object metadata when a file is uploaded. The checksum method in the S3 storage provider would query the metadata to return the appropriate value.